integration-tests/script/setup_k8s_cluster.sh

```diff
@@ -16,19 +16,13 @@
 
 set -e
 
-# Set Necessary ENV
-export CHANGE_MINIKUBE_NONE_USER=true
-export MINIKUBE_WANTUPDATENOTIFICATION=false
-export MINIKUBE_WANTREPORTERRORPROMPT=false
-export MINIKUBE_HOME=$HOME
+export INSTALL_K3S_VERSION=v1.21.14+k3s1
 export KUBECONFIG=$HOME/.kube/config
 
-sudo apt install -y conntrack
-
-# Lacunch K8S cluster
+# Launch K8S cluster
 curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.18.1/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
-curl -Lo minikube https://storage.googleapis.com/minikube/releases/v1.8.1/minikube-linux-amd64 && chmod +x minikube && sudo mv minikube /usr/local/bin/
-sudo /usr/local/bin/minikube start --profile=minikube --vm-driver=none --kubernetes-version=v1.18.1
-sudo /usr/local/bin/minikube update-context
-
+curl -sfL https://get.k3s.io | bash /dev/stdin  --docker
```

Comment:
The above download k3s for each test. That is probably a somewhat large task. Should the software be installed in the runner ahead of time? Can we do that? Is some K8s platform already available in the GHA runner?

Reply:
there is only one test. It seems that GHA runners have minikube installed but I was told that there were some errors. @tejaswini-imply knows what those errors were. 

I tried K3S and I felt that it's really lightweight to download and bring up. CI systems is one of the core use cases for K3S.

Reply:
Using minkube is causing these errors. sample run - https://github.com/tejaswini-imply/druid/actions/runs/3942544831/jobs/6748042868
```
stderr:
[573](https://github.com/tejaswini-imply/druid/actions/runs/3942544831/jobs/6748042868#step:5:574)
W0117 21:13:37.815530    2352 configset.go:202] WARNING: kubeadm cannot validate component configs for API groups [kubelet.config.k8s.io kubeproxy.config.k8s.io]
[574](https://github.com/tejaswini-imply/druid/actions/runs/3942544831/jobs/6748042868#step:5:575)
	[WARNING IsDockerSystemdCheck]: detected "cgroupfs" as the Docker cgroup driver. The recommended driver is "systemd". Please follow the guide at https://kubernetes.io/docs/setup/cri/
[575](https://github.com/tejaswini-imply/druid/actions/runs/3942544831/jobs/6748042868#step:5:576)
	[WARNING Swap]: running with swap on is not supported. Please disable swap
[576](https://github.com/tejaswini-imply/druid/actions/runs/3942544831/jobs/6748042868#step:5:577)
	[WARNING FileExisting-socat]: socat not found in system path
[577](https://github.com/tejaswini-imply/druid/actions/runs/3942544831/jobs/6748042868#step:5:578)
	[WARNING SystemVerification]: this Docker version is not on the list of validated versions: 20.10.22+azure-1. Latest validated version: 19.03
[578](https://github.com/tejaswini-imply/druid/actions/runs/3942544831/jobs/6748042868#step:5:579)
	[WARNING Service-Kubelet]: kubelet service is not enabled, please run 'systemctl enable kubelet.service'
[579](https://github.com/tejaswini-imply/druid/actions/runs/3942544831/jobs/6748042868#step:5:580)
W0117 21:13:56.868796    2352 manifests.go:225] the default kube-apiserver authorization-mode is "Node,RBAC"; using "Node,RBAC"
[580](https://github.com/tejaswini-imply/druid/actions/runs/3942544831/jobs/6748042868#step:5:581)
W0117 21:13:56.869683    2352 manifests.go:225] the default kube-apiserver authorization-mode is "Node,RBAC"; using "Node,RBAC"
[581](https://github.com/tejaswini-imply/druid/actions/runs/3942544831/jobs/6748042868#step:5:582)
error execution phase wait-control-plane: couldn't initialize a Kubernetes cluster
[582](https://github.com/tejaswini-imply/druid/actions/runs/3942544831/jobs/6748042868#step:5:583)
To see the stack trace of this error execute with --v=5 or higher
[583](https://github.com/tejaswini-imply/druid/actions/runs/3942544831/jobs/6748042868#step:5:584)

[584](https://github.com/tejaswini-imply/druid/actions/runs/3942544831/jobs/6748042868#step:5:585)
* 
[585](https://github.com/tejaswini-imply/druid/actions/runs/3942544831/jobs/6748042868#step:5:586)
* minikube is exiting due to an error. If the above message is not useful, open an issue:
[586](https://github.com/tejaswini-imply/druid/actions/runs/3942544831/jobs/6748042868#step:5:587)
  - https://github.com/kubernetes/minikube/issues/new/choose
[587](https://github.com/tejaswini-imply/druid/actions/runs/3942544831/jobs/6748042868#step:5:588)
Error:  Command execution failed.
```